sourcery_analytics/extractors.py

```diff
@@ -122,26 +126,30 @@ def from_function(
         """Construct an arbitrary extractor from a function of a node."""
         return cls(FunctionVisitor(function))
 
-    @functools.singledispatchmethod
     def extract(self, item: Extractable) -> typing.Iterator[T]:
         """Extract from source code, a node, a file, or directory."""
+        # `singledispatchmethod` confuses mypy, so wrap with a mypy-friendly interface
```

Comment:
**suggestion (docstrings):** Please update the docstring for function: `extract`

Reason for update: Introduction of a new extraction method implementation detail.

Suggested new docstring:
```
Extract from source code, a node, a file, or directory. This method now wraps a mypy-friendly interface for extraction.
```